### PR TITLE
chore(node-restart): Adding docs for node-restart experiment

### PR DESCRIPTION
**Pull request by ispeakc0de**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] Signed the commit for [DCO](https://github.com/litmuschaos/litmus-docs/blob/master/CONTRIBUTING.md#sign-your-work) check to be passed
-   [ ] Labelled this PR & related issue with `documentation` tag

### CODE REVIEW

File: docs/node-restart.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    <td> Defaults to `litmuschaos/go-runner:latest` </td>
+  </tr>
+  <tr>
+    <td> SSH_USER  </td>
+    <td> name of ssh user </td>
+    <td> Mandatory </td>
+    <td> Defaults to `root` </td>
+  </tr>
+  <tr>
+    <td> TARGET_NODES </td>
```
Comment (machacekondra) [inline docs/node-restart.md]: Unless I am missing something I think this won't work with the command separated list, but only as a single value.

File: docs/node-restart.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    <td> comma separated list of target nodes, subjected to chaos </td>
+    <td> Mandatory </td>
+    <td>  </td>
+  </tr>
+  <tr>
+    <td> TARGET_NODE_IPS </td>
+    <td> comma separated list of target node ips, subjected to chaos </td>
+    <td> Mandatory </td>
```
Comment (machacekondra) [inline docs/node-restart.md:152]: Unless I am missing something I think this won't work with the command separated list, but only as a single value.

File: docs/node-restart.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    <td> Restart the target node </td>
+    <td> GKE, EKS </td>
+  </tr>
+</table>
+
+## Prerequisites
+
+- Ensure that the Litmus Chaos Operator is running by executing `kubectl get pods` in operator namespace (typically, `litmus`). If not, install from [here](https://docs.litmuschaos.io/docs/getstarted/#install-litmus)
+- Ensure that the `node-restart` experiment resource is available in the cluster by executing `kubectl get chaosexperiments` in the desired namespace If not, install from [here](https://hub.litmuschaos.io/api/chaos/master?file=charts/generic/node-restart/experiment.yaml)
+- Ensure to create a Kubernetes secret having the rsa-key needed for ssh in the `CHAOS_NAMESPACE`. A sample secret file looks like:
```
Comment (machacekondra) [inline docs/node-restart.md]: I think `CHAOS_NAMESPACE` should be replaced by `TARGET_NODE`, if I understand the sentence correctly.
Maybe should be rephrased to something like:
```
Create a Kubernetes secret having the private SSH key for `SSH_USER` used to connect to `TARGET_NODE`
```

Comment (machacekondra) [reply docs/node-restart.md]: Also it's very important that the secret is named `id-rsa`, not sure if that should be specified somewhere in the doc. And the private SSH key data, named `ssh-privatekey`. It's correct in the example, but maybe would be nice to note it in the doc specifically.

File: docs/node-restart.md (hunk 1 above)
Comment (ispeakc0de) [reply docs/node-restart.md]: I made those modifications to my open PR but due to the increment of blast-radius. I rollback the changes to the `TARGET_NODE`(one node) only.

**Review by machacekondra — DISMISSED** [review]

File: docs/node-restart.md [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    <td> Generic </td>
+    <td> Restart the target node </td>
+    <td> GKE, EKS </td>
+  </tr>
+</table>
+
+## Prerequisites
+
+- Ensure that the Litmus Chaos Operator is running by executing `kubectl get pods` in operator namespace (typically, `litmus`). If not, install from [here](https://docs.litmuschaos.io/docs/getstarted/#install-litmus)
+- Ensure that the `node-restart` experiment resource is available in the cluster by executing `kubectl get chaosexperiments` in the desired namespace If not, install from [here](https://hub.litmuschaos.io/api/chaos/master?file=charts/generic/node-restart/experiment.yaml)
```
Comment (ksatchit) [inline docs/node-restart.md]:
```suggestion
- Ensure that the `node-restart` experiment resource is available in the cluster by executing `kubectl get chaosexperiments` in the desired namespace. If not, install from [here](https://hub.litmuschaos.io/api/chaos/master?file=charts/generic/node-restart/experiment.yaml)
```

File: docs/node-restart.md [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+-  Target Nodes should be in Ready state before chaos injection.
+
+## Exit-Criteria
+
+- Application pods should be healthy after chaos injection.
+- Target Nodes should be in Ready state after chaos injection.
+
+## Details
+
+-   Causes chaos to disrupt state of node for a certain chaos duration.
```
Comment (ksatchit) [inline docs/node-restart.md]:
```suggestion
-   Causes chaos to disrupt state of node by restarting it. 
```

(we don't really wait for duration)

File: docs/node-restart.md [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+
+## Details
+
+-   Causes chaos to disrupt state of node for a certain chaos duration. 
+-   Tests deployment sanity (replica availability & uninterrupted service) and recovery workflows of the application pod
+
+## Integrations
+
+-   Node Restart can be effected using the chaos library: `litmus`.
+-   The desired chaoslib can be selected by setting the above options as value for the env variable `LIB`
```
Comment (ksatchit) [inline docs/node-restart.md]: Right now we only have native support, so the second line can be omitted

File: docs/node-restart.md [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+- Create the ChaosEngine manifest prepared in the previous step to trigger the Chaos.
+
+  `kubectl apply -f chaosengine.yml`
+
+- If the chaos experiment is not executed, refer to the [troubleshooting](https://docs.litmuschaos.io/docs/faq-troubleshooting/) 
+  section to identify the root cause and fix the issues.
+
+### Watch Chaos progress
+
+- View the status of the nodess as they are subjected to node restart.
```
Comment (ksatchit) [inline docs/node-restart.md]:
```suggestion
- View the status of the nodes as they are subjected to node restart. 
```

**Comment by ksatchit** [comment]
Minor comments, looks good otherwise

**Review by ksatchit — APPROVED** [review]

**Review by uditgaurav — APPROVED** [review]

**Pull request merged**